mantle/kola/tests/misc/disk.go

```diff
@@ -0,0 +1,159 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package misc
+
+import (
+	"encoding/json"
+	"fmt"
+
+	ignv3types "github.com/coreos/ignition/v2/config/v3_0/types"
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform"
+	"github.com/coreos/mantle/platform/conf"
+	"github.com/coreos/mantle/util"
+)
+
+func init() {
+	register.RegisterTest(&register.Test{
+		Run:         varLibContainers,
+		ClusterSize: 0,
+		Name:        `misc.disk.varlibcontainers`,
+		Flags:       []register.Flag{},
+		Distros:     []string{"rhcos", "fcos"},
+		Platforms:   []string{"qemu-unpriv"},
+	})
+}
+
+func varLibContainers(c cluster.TestCluster) {
+	var m platform.Machine
+	var err error
+	disk := "disk1"
+	varMountName := "var-lib-containers.mount"
+	mkfsServiceName := fmt.Sprintf("systemd-mkfs@dev-disk-by-id-virtio-%s.service", disk)
+
+	options := platform.MachineOptions{
+		AdditionalDisks: []string{"1024M"},
+	}
+
+	mkfsUnit := fmt.Sprintf(`[Unit]
+	Description=Make File System on /dev/disk/by-id/%[1]s
+	DefaultDependencies=no
+	BindsTo=dev-disk-by\x2did-virtio\x2d%[1]s.device
+	After=dev-disk-by\x2did-virtio\x2d%[1]s.device var.mount
+	Before=systemd-fsck@dev-disk-by\x2did-virtio\x2d%[1]s.service
+	Before=shutdown.target
+
+	[Service]
+	Type=oneshot
+	RemainAfterExit=yes
+	ExecStart=/bin/bash -c "/bin/rm -rf /var/lib/containers/*"
+	ExecStart=mkfs.xfs /dev/disk/by-id/virtio-%[1]s
+	TimeoutSec=0
+
+	[Install]
+	WantedBy=var-lib-containers.mount`, disk)
+
+	varMountUnit := fmt.Sprintf(`[Unit]
+	Description=Mount %[2]s disk to /var/lib/containers
+	Before=local-fs.target
+	Requires=%[1]s
+	After=%[1]s
+
+	[Mount]
+	What=/dev/disk/by-id/virtio-%[2]s
+	Where=/var/lib/containers
+	Type=xfs
+	Options=defaults,prjquota
+
+	[Install]
+	WantedBy=local-fs.target`, mkfsServiceName, disk)
+
+	config := ignv3types.Config{
+		Ignition: ignv3types.Ignition{
+			Version: "3.0.0",
+		},
+		Storage: ignv3types.Storage{
+			Disks: []ignv3types.Disk{
+				{
+					Device:    "/dev/disk/by-id/virtio-disk1",
+					WipeTable: util.BoolToPtr(true),
+				},
+			},
+			Files: []ignv3types.File{
+				{
+					Node: ignv3types.Node{
+						Path: "/var/home/core/Dockerfile",
+					},
+					FileEmbedded1: ignv3types.FileEmbedded1{
+						Contents: ignv3types.FileContents{
+							Source: util.StrToPtr("data:,FROM%20scratch%0ACOPY%20os-release%20%2F"),
+						},
+						Mode: util.IntToPtr(420),
+					},
+				},
+			},
+		},
+		Systemd: ignv3types.Systemd{
+			Units: []ignv3types.Unit{
+				{
+					Name:     mkfsServiceName,
+					Contents: &mkfsUnit,
+					Enabled:  util.BoolToPtr(true),
+				},
+				{
+					Name:     varMountName,
+					Contents: &varMountUnit,
+					Enabled:  util.BoolToPtr(false),
+				},
+			},
+		},
+	}
+
+	buf, err := json.Marshal(config)
```

Comment:
Why not just make this `serailizedConfig, err := ...` rather than setting `serializedConfig := buf` after the error check?